scripts/test-tlsnotary-proxy.ts

```
@@ -0,0 +1,223 @@
+// doesn't.
+const fetchInit: RequestInit = ALLOW_INSECURE
+    ? ({ tls: { rejectUnauthorized: false } } as never)
+    : {}
+
+// Sanitize log message to strip ANSI escapes + control chars + cap
+// length. The driver echoes upstream server text into the terminal;
+// without this, a malicious node response could inject terminal
+// escape sequences (CWE-117). Sonar tssecurity:S5145.
+function sanitize(s: string): string {
+    return s
+        .replace(/[\x00-\x1f\x7f]/g, "?")
```

ESLint

[error] Unexpected control character(s) in regular expression: \x00, \x1f.

(no-control-regex)